testutils/clustermanager/gke.go

```diff
@@ -44,22 +47,29 @@ type GKECluster struct {
 // region: default to regional cluster if not provided, and use default backup regions
 // zone: default is none, must be provided together with region
 func (gs *GKEClient) Setup(numNodes *int, nodeType *string, region *string, zone *string, project *string) (ClusterOperations, error) {
-	var err error
-	gc := &GKECluster{}
-	if nil != project { // use provided project and create cluster
+	gc := &GKECluster{
+		Exec: standardExec,
+	}
+	// check for local run
+	if nil != project { // if project is supplied, use it and create cluster
 		gc.Project = project
 		gc.NeedCleanup = true
-	} else if err = gc.checkEnvironment(); nil != err {
+	} else if err := gc.checkEnvironment(); nil != err {
 		return nil, fmt.Errorf("failed checking existing cluster: '%v'", err)
-	}
-	if nil != gc.Cluster {
+	} else if nil != gc.Cluster { // return if Cluster was already set by kubeconfig
 		return gc, nil
 	}
+	// check for Prow
 	if common.IsProw() {
-		if *gc.Project, err = boskos.AcquireGKEProject(); nil != err {
+		if project, err := boskos.AcquireGKEProject(); nil != err {
 			return nil, fmt.Errorf("failed acquire boskos project: '%v'", err)
+		} else if "" != project {
+			gc.Project = &project
 		}
 	}
+	if nil == gc.Project {
```

Comment:
Does it catch empty string too, as you mentioned above?

Reply:
good catch, add check here and simplifies above